src/prefect/orion/models/block_documents.py

```diff
@@ -445,14 +446,28 @@ async def update_block_document(
         # realizing they are posting back obfuscated data, so we disregard the update
         flat_update_data = dict_to_flatdict(update_values["data"])
         flat_current_data = dict_to_flatdict(current_data)
-        for field in current_block_document.block_schema.fields.get(
+        for secret_field in current_block_document.block_schema.fields.get(
             "secret_fields", []
         ):
-            key = tuple(field.split("."))
-            current_secret = flat_current_data.get(key)
+            secret_key = tuple(secret_field.split("."))
+            current_secret = flat_current_data.get(secret_key)
             if current_secret is not None:
-                if flat_update_data.get(key) == obfuscate_string(current_secret):
-                    del flat_update_data[key]
+                if flat_update_data.get(secret_key) == obfuscate_string(current_secret):
+                    del flat_update_data[secret_key]
+            # Looks for obfuscated values nested under a secret field with a wildcard.
+            # If any obfuscated values are found, we assume that it shouldn't be update,
+            # and they are replaced with the current value for that key to avoid losing
+            # data during update.
+            elif "*" in secret_key:
+                wildcard_index = secret_key.index("*")
+                for data_key in flat_update_data.keys():
+                    if secret_key[0:wildcard_index] == data_key[
+                        0:wildcard_index
+                    ] and flat_update_data[data_key] == obfuscate(
+                        flat_update_data[data_key]
+                    ):
```

Comment:
Black formats these so weird. Generally, adding parens will make it more readable.